File: Ruby/c_rank/word_check.rb
score = 0

len = gets.to_i

len.times do |i|
  ans, target = gets.chomp.split(" ")

  # 完全一致だった場合
  if ans == target
    score += 2
    next
  end

  ans_array = ans.split("")
  target_array = target.split("")

  # 文字数が不一致だった場合はloopをスキップ
  if ans_array.size != target_array.size
    next
  end
  
  # 文字の一致数をカウント
  count = 0
  (ans_array.size).times do |num|
    if ans_array[num] == target_array[num]
      count += 1
    end
  end

  # カウントによってスコアに加算
  if count == ans_array.size
    score += 2
  elsif count == ans_array.size - 1
    score += 1
  end
end

# 結果表示
puts score

File: java/c_rank/WordCheck.java
package java.c_rank;

import java.util.Scanner;

public class WordCheck {

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    // 問題の数
    int n = sc.nextInt();
    int score = 0;

    // 問題の数for文を回す
    for (int i = 0; i < n; i++) {
      String ans = sc.next();
      String correct = sc.next();

      // // デバック
      // System.out.println(ans);
      // System.out.println(correct);
      // System.out.println(ans == correct);

      // 文字の数が合わなければスキップ
      if (ans.length() != correct.length()) {
        continue;
      }

      // 完全一致でscore+2
      if (ans.equals(correct)) {
        score += 2;
        continue;
      }

      // 文字の一致数を確認するための配列を生成
      String[] ansArray = new String[ans.length()];
      String[] correctArray = new String[correct.length()];

      // 配列に格納
      ansArray = ans.split("");
      correctArray = correct.split("");

      // 一致数の確認
      int count = 0;
      for (int j = 0; j < ansArray.length; j++) {
        if (ansArray[j].equals(correctArray[j])) {
          count += 1;
        }
      }

      // 1文字違いならscore+1
      if (count == ansArray.length - 1) {
        score += 1;
      }
    }

    // 結果表示
    System.out.println(score);
    sc.close();
  }
}
